h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderBuilder.java

```diff
@@ -36,8 +36,9 @@ public void init(boolean expensive) {
   private class TargetEncoderDriver extends Driver {
     @Override
     public void computeImpl() {
-      final String[] encoded_columns = Frame.VecSpecifier.vecNames(_parms._encoded_columns);
-      TargetEncoder tec = new TargetEncoder(encoded_columns);
+      final String[] encodedColumns = encodedColumnsFromIgnored(_parms.train(), _parms._ignored_columns,
```

Comment:
This doesn't look correct to me. At this point the Frame should only contain columns that will actually be used, it shouldn't skip any additional columns. You should also never work with `_parms.train()` here because it is a different frame that the one that was adapted by the model builder initialization.

There is a specific mechanism for column fitlering already this one should be used.

Reply:
It's not correct and I've fixed that. There is no need to do this manually - just use the names from the already adapted training frame in ModelBuilder's init and remove fold column name and response column name.

Reply:
@michalkurka `There is a specific mechanism for column filtering already this one should be used.` Do you mean that it should be done in `TargetEncoderBuilder.init()`? Would be the right place imo.
@Pscheidl why did you completely remove the logic? I don't think that the non-categorical columns are excluded anymore.
Wouldn't it be fine if it were just moved to `init`, overriding `_parms. _encoded_columns`?
By the way:
```
  public TargetEncoderBuilder(TargetEncoderModel.TargetEncoderParameters parms) {
    super(parms);
    super.init(false);
  }
```
should be :
```
  public TargetEncoderBuilder(TargetEncoderModel.TargetEncoderParameters parms) {
    super(parms);
    init(false);
  }
```
this could lead to bugs if overriding `init`.

Reply:
> Do you mean that it should be done in TargetEncoderBuilder.init()?

@seb-h2o yes, it should be done as a part of the init method, specifically, I meant to override `ignoreBadColumns` which is supposed to drop columns that are not relevant to the particular algo (in this case we also need to drop the numeric columns).

Reply:
`ignoredBadColumns`: nice, never noticed this method.
Yes, could definitely be overridden in `TargetEncoderBuilder`: @Pscheidl

Reply:
@michalkurka the thing is: we still want the numeric columns in the output, so we don't want to remove them from train, we just want to remove them from `encoded_columns` param.
So `init` should be a better place for this.

Reply:
actually nowhere in the current code do I see anything close to:
```
encoded_columns = categorical train columns not in ignored_columns
```
not the "categorical train columns" part, 
nor the "not in ignored_columns" part.
So I'm even confused how it even works when used from  clients currently (as they provide only `ignored_columns`)

Reply:
> @michalkurka the thing is: we still want the numeric columns in the output, so we don't want to remove them from train, we just want to remove them from encoded_columns param.

Hmm, okay... if the goal is to leave the dataset intact then I think we also need to defeat the default login of `ignoreBadColumns`.

Why does TE need to know about columns that are not being used for training? IMHO they should just be dropped.

Reply:
@michalkurka sorry, forget my previous post, I'm afraid I got confused again with this part. We don't need those columns for training, we just want them in the predict/transform result, and this should work anyway if they are removed from `train` before training.
So back to your suggestion: `ignoredBadColumns` should be overridden in `TargetEncoderBuilder` to remove non-numeric columns.